cli/pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,158 @@
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/rilldata/rill/cli/pkg/config"
+	"github.com/rilldata/rill/cli/pkg/dotrill"
+)
+
+type Telemetry struct {
+	Enabled     bool
+	InstallID   string
+	Version     string
+	BuildCommit string
+	BuildTime   string
+	IsDev       bool
+	authHeader  string
+	events      [][]byte
+}
+
+const (
+	RillIntakeURL      = "https://intake.rilldata.io/events/data-modeler-metrics"
+	RillIntakeUser     = "data-modeler"
+	RillIntakePassword = "lkh8T90ozWJP/KxWnQ81PexRzpdghPdzuB0ly2/86TeUU8q/bKiVug==" //nolint:gosec //Need to figure out a way to add this during build time.
```

Comment:
Let's just keep it here, it's a lot easier. But maybe change "Password" to "Key" to make it not sound confidential, haha?